PSL/SENSORS/VL531X.py

```diff
@@ -0,0 +1,451 @@
+_GLOBAL_CONFIG_VCSEL_WIDTH = 0x32
+_GLOBAL_CONFIG_SPAD_ENABLES_REF_0 = 0xB0
+_GLOBAL_CONFIG_SPAD_ENABLES_REF_1 = 0xB1
+_GLOBAL_CONFIG_SPAD_ENABLES_REF_2 = 0xB2
+_GLOBAL_CONFIG_SPAD_ENABLES_REF_3 = 0xB3
+_GLOBAL_CONFIG_SPAD_ENABLES_REF_4 = 0xB4
+_GLOBAL_CONFIG_SPAD_ENABLES_REF_5 = 0xB5
+_GLOBAL_CONFIG_REF_EN_START_SELECT = 0xB6
+_DYNAMIC_SPAD_NUM_REQUESTED_REF_SPAD = 0x4E
+_DYNAMIC_SPAD_REF_EN_START_OFFSET = 0x4F
+_POWER_MANAGEMENT_GO1_POWER_FORCE = 0x80
+_VHV_CONFIG_PAD_SCL_SDA__EXTSUP_HV = 0x89
+_ALGO_PHASECAL_LIM = 0x30
+_ALGO_PHASECAL_CONFIG_TIMEOUT = 0x30
+_VCSEL_PERIOD_PRE_RANGE = 0
+_VCSEL_PERIOD_FINAL_RANGE = 1
+
+_ID_REGISTERS = (
+    (0xC0, 0xEE),
+    (0xC1, 0xAA),
+    (0xC2, 0x10),
+)
+
+_SPAD_CONFIG = (
+    (0xFF, 0x01),
+    (_DYNAMIC_SPAD_REF_EN_START_OFFSET, 0x00),
+    (_DYNAMIC_SPAD_NUM_REQUESTED_REF_SPAD, 0x2C),
+    (0xFF, 0x00),
+    (_GLOBAL_CONFIG_REF_EN_START_SELECT, 0xB4),
+)
+
+# TODO: What do these values mean?
+_TUNING_CONFIG = (
+    (0xFF, 0x01),
+    (0x00, 0x00),
+    (0xFF, 0x00),
+    (0x09, 0x00),
+    (0x10, 0x00),
+    (0x11, 0x00),
+    (0x24, 0x01),
+    (0x25, 0xFF),
+    (0x75, 0x00),
+    (0xFF, 0x01),
+    (0x4E, 0x2C),
+    (0x48, 0x00),
+    (0x30, 0x20),
+    (0xFF, 0x00),
+    (0x30, 0x09),
+    (0x54, 0x00),
+    (0x31, 0x04),
+    (0x32, 0x03),
+    (0x40, 0x83),
+    (0x46, 0x25),
+    (0x60, 0x00),
+    (0x27, 0x00),
+    (0x50, 0x06),
+    (0x51, 0x00),
+    (0x52, 0x96),
+    (0x56, 0x08),
+    (0x57, 0x30),
+    (0x61, 0x00),
+    (0x62, 0x00),
+    (0x64, 0x00),
+    (0x65, 0x00),
+    (0x66, 0xA0),
+    (0xFF, 0x01),
+    (0x22, 0x32),
+    (0x47, 0x14),
+    (0x49, 0xFF),
+    (0x4A, 0x00),
+    (0xFF, 0x00),
+    (0x7A, 0x0A),
+    (0x7B, 0x00),
+    (0x78, 0x21),
+    (0xFF, 0x01),
+    (0x23, 0x34),
+    (0x42, 0x00),
+    (0x44, 0xFF),
+    (0x45, 0x26),
+    (0x46, 0x05),
+    (0x40, 0x40),
+    (0x0E, 0x06),
+    (0x20, 0x1A),
+    (0x43, 0x40),
+    (0xFF, 0x00),
+    (0x34, 0x03),
+    (0x35, 0x44),
+    (0xFF, 0x01),
+    (0x31, 0x04),
+    (0x4B, 0x09),
+    (0x4C, 0x05),
+    (0x4D, 0x04),
+    (0xFF, 0x00),
+    (0x44, 0x00),
+    (0x45, 0x20),
+    (0x47, 0x08),
+    (0x48, 0x28),
+    (0x67, 0x00),
+    (0x70, 0x04),
+    (0x71, 0x01),
+    (0x72, 0xFE),
+    (0x76, 0x00),
+    (0x77, 0x00),
+    (0xFF, 0x01),
+    (0x0D, 0x01),
+    (0xFF, 0x00),
+    (0x80, 0x01),
+    (0x01, 0xF8),
+    (0xFF, 0x01),
+    (0x8E, 0x01),
+    (0x00, 0x01),
+    (0xFF, 0x00),
+    (0x80, 0x00),
+)
+
+# I2C general - probably implemented in firmware; not our concern here
+#  # datasheet p18: first byte is address and read/write direction
+#  WRITE_ADDRESS = 0x52
+#  READ_ADDRESS = 0x53
+
+# abbreviations and acronyms
+# VCSEL - Vertical Cavity Surface Emitting Laser
+# SPAD - Single Photon Avalanche Diode
+# MSRC - Minimum Signal Rate Check
+class VL531X():
+    PLOTNAMES = ['Data']
+    ADDRESS = 0x29
+    name = 'Time-of-Flight Proximity Sensor'
+
+    def __init__(self, handler, **args):
```

Comment:
Double star arguments should be called `**kwargs`. Avoid entirely if possible, use explicit arguments instead.